nextjs-app/app/services/likesAndDislikesService/likesAndDislikes.ts

```
@@ -0,0 +1,97 @@
+
+enum VoteType {
+    LIKE = "like",
+    DISLIKE = "dislike",
+    LIKED = "liked",
+    DISLIKED = "disliked"
+}
+
+interface PlayerVote {
+    userSteamId: string;
+    otherPlayerSteamId: string;
+    type: VoteType;
+}
+
+export async function putLikesAndDislikes({ userSteamId, otherPlayerSteamId, type }: PlayerVote) {
+    const db = await getDbInstance();
+    try {
+        
+        await new Promise((resolve, reject) => db.run("BEGIN TRANSACTION", (err) => (err ? reject(err) : resolve(true))));
+        
+        const existingVote: Array<Record<string, any>> = await new Promise((resolve, reject) => {
+            db.all(
+                `SELECT likes_dislikes FROM likeDislike WHERE steam_id = ? AND other_player_steam_id = ?`,
+                [String(userSteamId), String(otherPlayerSteamId)],
+                (err, rows) => {
+                    if (err) {
+                        console.error("Error executing query:", err);
+                        return reject(err);
+                    }
+                    resolve(rows as any);
+                }
+            );
+        });
+
+        if (type === VoteType.LIKED || type === VoteType.DISLIKED) {
+            // If type is liked or disliked, set likes_dislikes to NULL
+            await new Promise((resolve, reject) => {
+                db.run(
+                    `UPDATE likeDislike SET likes_dislikes = NULL WHERE steam_id = ? AND other_player_steam_id = ?`,
+                    [String(userSteamId), String(otherPlayerSteamId)],
+                    function (err) {
+                        if (err) {
+                            console.error("Error updating database:", err);
+                            return reject(err);
+                        }
+                        resolve(true);
+                    }
+                );
+            });
+        } else {
+            const likeDislikeValue = type === VoteType.LIKE ? 1 : 0;
+            if (existingVote.length > 0) {
```

if we are trying to like/dislike a player, and it is already liked/disliked (vote exists) that's also invalid case, and we should throw an error

---

Added